msteams-platform/concepts/messaging-extensions/create-extensions.md

```diff
@@ -110,6 +110,162 @@ To initiate actions from a  messaging extension set the `type` parameter to `act
+    "canUpdateConfiguration": true,
+    "commands": [
+      {
+        "id": "reassignTodo",
+        "description": "Reassign a todo item",
+        "title": "Create To Do",
+        "type": "Action",
+        "context": ["message"],
+        "fetchTask": true
+    }]
+    ...
+
+```
+
+Below is an example of the request that will be sent to your bot.
+
+```json
+{
+  "name": "composeExtension/submitAction",
+  "type": "invoke",
+  "timestamp": "2019-02-26T01:01:24.509Z",
+  "localTimestamp": "2019-02-26T01:01:24.509Z",
+  "entities": [
+    {
+      "locale": "en-US",
+      "country": "US",
+      "platform": "Windows",
+      "timezone": "America/Los_Angeles",
+      "type": "clientInfo"
+    }
+  ],
+  "value": {
+    "commandId": "setReminder",
+    "commandContext": "message",
+    "messagePayload": {
+      "id": "1111111111",
+      "replyToId": null,
+      "createdDateTime": "2019-02-25T21:29:36.065Z",
+      "lastModifiedDateTime": null,
+      "deleted": false,
+      "subject": "Message subject",
+      "summary": null,
+      "importance": "normal",
+      "locale": "en-us",
+      "body": {
+        "contentType": "html",
+        "content": "this is the message"
+		},
+      "from": {
+        "device": null,
+        "conversation": null,
+        "user": {
+          "userIdentityType": "aadUser",
+          "id": "wxyz12ab8-ab12-cd34-ef56-098abc123876",
+          "displayName": "Jamie Smythe"
+        },
+        "application": null
+      },
+      "reactions": [
+        {
+          "reactionType": "like",
+          "createdDateTime": "2019-02-25T22:40:40.806Z",
+          "user": {
+            "device": null,
+            "conversation": null,
+            "user": {
+              "userIdentityType": "aadUser",
+              "id": "qrst12346-ab12-cd34-ef56-098abc123876",
+              "displayName": "Jim Brown"
+            },
+            "application": null
+          }
+        }
+      ],
+      "mentions": [
+        {
+          "id": 0,
+          "mentionText": "Sarah",
+          "mentioned": {
+            "device": null,
+            "conversation": null,
+            "user": {
+              "userIdentityType": "aadUser",
+              "id": "ab12345678-ab12-cd34-ef56-098abc123876",
+              "displayName": "Sarah"
+            },
+            "application": null
+          }
+        }
+      ]
+    },
+    "context": {
+      "theme": "default"
+    },
+    "data": {
+      "dateChoice": "15-sec",
+      "customDeltaHour": "",
+      "customDeltaMin": "",
+      "customDeltaSec": ""
+    }
+  },
+  "textLocale": "en-US",
+  "sourceEvent": {
+    "channel": {
+      "id": "11:123413432123412341234@thread.skype"
+    },
+    "team": {
+      "id": "11:something_asdfadf12341234dasdf1341@thread.skype"
+    },
+    "tenant": {
+      "id": "12345ab8-ab12-cd34-ef56-098abc123876"
+    },
+    "source": {
+      "name": "compose"
+    }
+  },
+  "text": "",
+  "attachments": [],
+  "address": {
```

Comment:
@clearab Just checking, is this the serialized IMessage in a Node app, or the actual JSON payload of a request? The "address" and "source" properties make me think this is the serialized IMessage. I think the raw incoming payload is more useful.

Reply:
updated the JSON example to focus on just the value object that contains the message details